baremetal/metal3machine_manager.go

```diff
@@ -775,14 +864,15 @@ func (m *MachineManager) chooseHost(ctx context.Context) (*bmh.BareMetalHost, *p
 	labelSelector = labelSelector.Add(reqs...)
 
 	availableHosts := []*bmh.BareMetalHost{}
+	availableHostsWithNodeReuse := []*bmh.BareMetalHost{}
 
 	for i, host := range hosts.Items {
 		if host.Spec.ConsumerRef != nil && consumerRefMatches(host.Spec.ConsumerRef, m.Metal3Machine) {
 			m.Log.Info("Found host with existing ConsumerRef", "host", host.Name)
 			helper, err := patch.NewHelper(&hosts.Items[i], m.client)
 			return &hosts.Items[i], helper, err
 		}
-		if host.Spec.ConsumerRef != nil {
+		if host.Spec.ConsumerRef != nil || (m.nodeReuseLabelExists(ctx, &host) && !m.nodeReuseLabelMatches(ctx, &host)) {
```

Comment:
can we break the conditions in multiple lines for better readability ?

Reply:
Done